### PR TITLE
docs(reply): specify streams content-type

### PR DESCRIPTION
**Pull request by D10f**
…pecified manually when sending back streams.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/Reference/Reply.md [hunk 1]
```diff
@@ -656,13 +656,14 @@ fastify.get('/json', options, function (request, reply) {
 #### Streams
 <a id="send-streams"></a>
 
-*send* can also handle streams out of the box. If you are sending a stream and
-you have not set a `'Content-Type'` header, *send* will set it at
-`'application/octet-stream'`.
+*send* can also handle streams by setting the `'Content-Type'` header to
+`'application/octet-stream'`. For proxied responses, this header should be
```
Comment (mcollina) [inline docs/Reference/Reply.md:660]: I don't understand the reference to proxied responses, could you clarify?

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by D10f** [comment]
I added it based on the original issue that explains why streams should not have a content-type header by default: [#3085](https://github.com/fastify/fastify/issues/3085)

> (...) fastify is being used as a proxy and is receiving an empty stream response.

To be honest I wasn't really sure if this should be mentioned, since this part of the documentation is already specific about stream responses. Perhaps is best to not mention this to avoid confusion and only point out that stream responses need the proper header.

**Comment by mcollina** [comment]
I think so, avoid mentioning the proxy. Thanks!

File: docs/Reference/Reply.md (hunk 1 above)
Comment (Uzlopak) [reply docs/Reference/Reply.md:660]:
```suggestion
`'application/octet-stream'`.
```

File: docs/Reference/Reply.md [hunk 2]
```diff
@@ -656,13 +656,14 @@ fastify.get('/json', options, function (request, reply) {
 #### Streams
 <a id="send-streams"></a>
 
-*send* can also handle streams out of the box. If you are sending a stream and
-you have not set a `'Content-Type'` header, *send* will set it at
-`'application/octet-stream'`.
+*send* can also handle streams by setting the `'Content-Type'` header to
+`'application/octet-stream'`. For proxied responses, this header should be 
+unspecified instead.
```
Comment (Uzlopak) [inline docs/Reference/Reply.md:661]:
```suggestion
```

**Review by Uzlopak — APPROVED** [review]

**Review by Fdawgs — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**